test/e2e/helpers_test.go

```diff
@@ -346,55 +369,69 @@ func deployIngress(ctx context.Context, t *testing.T, env environments.Environme
 			},
 		},
 		Route: &kongv1.KongIngressRoute{
-			Methods: []*string{lo.ToPtr("GET")},
+			Methods: []*string{lo.ToPtr(http.MethodGet)},
 		},
 	}
 	_, err = c.ConfigurationV1().KongIngresses(corev1.NamespaceDefault).Create(ctx, king, metav1.CreateOptions{})
 	require.NoError(t, err)
 	t.Logf("creating an ingress for service %s with ingress.class %s", service.Name, ingressClass)
-	ingress := generators.NewIngressForService("/httpbin", map[string]string{
+	ingress := generators.NewIngressForService("/echo", map[string]string{
 		annotations.IngressClassKey: ingressClass,
 		"konghq.com/strip-path":     "true",
 		"konghq.com/override":       kongIngressName,
 	}, service)
 	require.NoError(t, clusters.DeployIngress(ctx, env.Cluster(), corev1.NamespaceDefault, ingress))
 }
 
-func verifyIngress(ctx context.Context, t *testing.T, env environments.Environment) {
+func verifyIngressWithEchoBackends(ctx context.Context, t *testing.T, env environments.Environment) {
 	t.Helper()
 
-	t.Log("finding the kong proxy service ip")
-	proxyIP := getKongProxyIP(ctx, t, env)
+	t.Log("finding the service URL (through Kong proxy service ip)")
+	echoURL := fmt.Sprintf("http://%s/echo", getKongProxyIP(ctx, t, env))
 
-	t.Logf("waiting for route from Ingress to be operational at http://%s/httpbin", proxyIP)
-	require.Eventually(t, func() bool {
-		resp, err := helpers.DefaultHTTPClient().Get(fmt.Sprintf("http://%s/httpbin", proxyIP))
-		if err != nil {
-			return false
+	t.Logf(
+		"waiting for route from Ingress to be operational at %s and forward traffic to all %d backends",
+		echoURL, numberOfEchoBackends,
+	)
+	uniqueResponses := make(map[string]struct{})
+	require.EventuallyWithT(t, func(c *assert.CollectT) {
+		resp, err := helpers.DefaultHTTPClient().Get(echoURL)
+		if !assert.NoError(c, err) {
+			return
 		}
 		defer resp.Body.Close()
-		if resp.StatusCode == http.StatusOK {
-			b := new(bytes.Buffer)
-			_, err := b.ReadFrom(resp.Body)
-			if err != nil {
-				return false
-			}
-			if !strings.Contains(b.String(), "<title>httpbin.org</title>") {
-				return false
-			}
-		} else {
-			return false
+
+		if !assert.Equal(c, http.StatusOK, resp.StatusCode) {
+			return
+		}
+		b := new(bytes.Buffer)
+		_, err = b.ReadFrom(resp.Body)
+		if !assert.NoError(c, err) {
+			return
+		}
+		// Every backend responds with its own (different) IP address.
+		if msg := b.String(); strings.Contains(msg, "With IP address") {
+			uniqueResponses[msg] = struct{}{}
 		}
-		// verify the KongIngress method restriction
+		assert.Len(c, uniqueResponses, numberOfEchoBackends)
+	},
+		ingressWait, 100*time.Millisecond,
```

Comment:
I checked it manually and propagation of EndpointSlices took more closely to ~1-2s, but it may vary so checking more often is not a problem. I'll adjust